public/app/features/dashboard/components/PanelEditor/PanelEditorTabs.tsx

```diff
@@ -16,55 +16,77 @@ interface PanelEditorTabsProps {
   onChangeTab: (tab: PanelEditorTab) => void;
 }
 
-export const PanelEditorTabs: React.FC<PanelEditorTabsProps> = ({ panel, dashboard, tabs, onChangeTab }) => {
-  const styles = getPanelEditorTabsStyles();
-  const activeTab = tabs.find(item => item.active)!;
+export class PanelEditorTabs extends PureComponent<PanelEditorTabsProps> {
+  componentDidMount() {
+    const { panel } = this.props;
+    panel.on(CoreEvents.queryAdded, this.triggerForceUpdate);
+    panel.on(CoreEvents.queryRemoved, this.triggerForceUpdate);
+    panel.on(CoreEvents.transformationAdded, this.triggerForceUpdate);
+    panel.on(CoreEvents.transformationRemoved, this.triggerForceUpdate);
+  }
```

Comment:
If this still would have been a functional component we could have used a `useEffect` hook to handle the unsubscribe of the events.

But in this case we might be able to do something like:

```suggestion
  componentDidMount() {
      const { panel } = this.props;
      this.subscriptions = [
          panel.on(CoreEvents.queryAdded, this.triggerForceUpdate),
          panel.on(CoreEvents.queryRemoved, this.triggerForceUpdate),
          panel.on(CoreEvents.transformationAdded, this.triggerForceUpdate),
          panel.on(CoreEvents.transformationRemoved, this.triggerForceUpdate),
      ];
  }
```

Reply:
I am assuming that the `panel.on` is returning a function to unsubscribe, but that might not be the case?